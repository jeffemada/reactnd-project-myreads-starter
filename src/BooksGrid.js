import PropTypes from 'prop-types';
import React from 'react';
import Book from './Book';

function BooksGrid(props) {
  const { books, onRefresh } = props;

  return (
    <ol className="books-grid">
      {books.map((book) => (
        <Book key={book.id} book={book} onRefresh={onRefresh} />
      ))}
    </ol>
  );
}

BooksGrid.propTypes = {
  books: PropTypes.array.isRequired,
  onRefresh: PropTypes.func.isRequired
};

export default BooksGrid;
