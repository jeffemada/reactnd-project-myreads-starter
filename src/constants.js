export const SHELVES = [
  { id: "wantToRead", title: "Want to Read" },
  { id: "currentlyReading", title: "Currently Reading" },
  { id: "read", title: "Read" },
  { id: "none", title: "None" }
];
