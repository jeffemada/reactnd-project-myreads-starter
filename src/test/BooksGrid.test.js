import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import BooksGrid from '../BooksGrid';
import { SEARCH_BOOKS_MOCK } from './constants';

Enzyme.configure({ adapter: new Adapter() });

it('should render a BooksGrid', () => {
  const wrapper = shallow(<BooksGrid books={SEARCH_BOOKS_MOCK} onRefresh={jest.fn()} onLoading={jest.fn()} />);
  expect(wrapper).toMatchSnapshot();
});
