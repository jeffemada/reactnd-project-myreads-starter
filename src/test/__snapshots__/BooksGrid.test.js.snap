// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a BooksGrid 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BooksGrid
    books={
      Array [
        Object {
          "authors": Array [
            "Divya Manian",
          ],
          "id": "aaHXYmZLYNQC",
          "imageLinks": Object {
            "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
            "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
          },
          "shelf": "wantToRead",
          "title": "HTML5 Boilerplate Web Development",
        },
        Object {
          "authors": Array [
            "Julie Ratner",
          ],
          "id": "sKvi5fVL-uAC",
          "imageLinks": Object {
            "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
            "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
          },
          "title": "Human Factors and Web Development",
        },
      ]
    }
    onLoading={[MockFunction]}
    onRefresh={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Book
          book={
            Object {
              "authors": Array [
                "Divya Manian",
              ],
              "id": "aaHXYmZLYNQC",
              "imageLinks": Object {
                "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
              },
              "shelf": "wantToRead",
              "title": "HTML5 Boilerplate Web Development",
            }
          }
          onLoading={[MockFunction]}
          onRefresh={[MockFunction]}
        />,
        <Book
          book={
            Object {
              "authors": Array [
                "Julie Ratner",
              ],
              "id": "sKvi5fVL-uAC",
              "imageLinks": Object {
                "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
              },
              "title": "Human Factors and Web Development",
            }
          }
          onLoading={[MockFunction]}
          onRefresh={[MockFunction]}
        />,
      ],
      "className": "books-grid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "aaHXYmZLYNQC",
        "nodeType": "function",
        "props": Object {
          "book": Object {
            "authors": Array [
              "Divya Manian",
            ],
            "id": "aaHXYmZLYNQC",
            "imageLinks": Object {
              "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
              "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
            },
            "shelf": "wantToRead",
            "title": "HTML5 Boilerplate Web Development",
          },
          "onLoading": [MockFunction],
          "onRefresh": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "sKvi5fVL-uAC",
        "nodeType": "function",
        "props": Object {
          "book": Object {
            "authors": Array [
              "Julie Ratner",
            ],
            "id": "sKvi5fVL-uAC",
            "imageLinks": Object {
              "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
              "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
            },
            "title": "Human Factors and Web Development",
          },
          "onLoading": [MockFunction],
          "onRefresh": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ol",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Book
            book={
              Object {
                "authors": Array [
                  "Divya Manian",
                ],
                "id": "aaHXYmZLYNQC",
                "imageLinks": Object {
                  "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                  "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                },
                "shelf": "wantToRead",
                "title": "HTML5 Boilerplate Web Development",
              }
            }
            onLoading={[MockFunction]}
            onRefresh={[MockFunction]}
          />,
          <Book
            book={
              Object {
                "authors": Array [
                  "Julie Ratner",
                ],
                "id": "sKvi5fVL-uAC",
                "imageLinks": Object {
                  "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                  "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
                },
                "title": "Human Factors and Web Development",
              }
            }
            onLoading={[MockFunction]}
            onRefresh={[MockFunction]}
          />,
        ],
        "className": "books-grid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "aaHXYmZLYNQC",
          "nodeType": "function",
          "props": Object {
            "book": Object {
              "authors": Array [
                "Divya Manian",
              ],
              "id": "aaHXYmZLYNQC",
              "imageLinks": Object {
                "smallThumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                "thumbnail": "http://books.google.com/books/content?id=aaHXYmZLYNQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
              },
              "shelf": "wantToRead",
              "title": "HTML5 Boilerplate Web Development",
            },
            "onLoading": [MockFunction],
            "onRefresh": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "sKvi5fVL-uAC",
          "nodeType": "function",
          "props": Object {
            "book": Object {
              "authors": Array [
                "Julie Ratner",
              ],
              "id": "sKvi5fVL-uAC",
              "imageLinks": Object {
                "smallThumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
                "thumbnail": "http://books.google.com/books/content?id=sKvi5fVL-uAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
              },
              "title": "Human Factors and Web Development",
            },
            "onLoading": [MockFunction],
            "onRefresh": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ol",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
